pkg/controllers/dynakube/extension/service.go

```diff
@@ -0,0 +1,92 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/service"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (r *reconciler) reconcileService(ctx context.Context) error {
+	if r.dk.PrometheusEnabled() {
+		return r.createOrUpdateService(ctx)
+	} else {
+		if meta.FindStatusCondition(*r.dk.Conditions(), extensionsServiceConditionType) == nil {
+			return nil
+		}
+		defer meta.RemoveStatusCondition(r.dk.Conditions(), extensionsServiceConditionType)
+
+		svc, err := r.buildService()
+		if err != nil {
+			log.Error(err, "could not build service during cleanup")
+			conditions.SetServiceGenFailed(r.dk.Conditions(), extensionsServiceConditionType, err)
```

Comment:
doesn't do anything, as the defer will just remove the condition

Reply:
Good catch!

Reply:
fixed in d19559f5